lib/ClangImporter/ImportType.cpp

```diff
@@ -1435,12 +1452,43 @@ Type ClangImporter::Implementation::importType(clang::QualType type,
   SwiftTypeConverter converter(*this, allowNSUIntegerAsInt, bridging);
   auto importResult = converter.Visit(type);
 
-  // Now fix up the type based on we're concretely using it.
-  return adjustTypeForConcreteImport(*this, type, importResult.AbstractType,
-                                     importKind, importResult.Hint,
-                                     allowNSUIntegerAsInt, bridging,
-                                     optionality,
-                                     resugarNSErrorPointer);
+  // Now fix up the type based on how we're concretely using it.
+  Type adjustedType;
+  bool isIUO;
+  std::tie(adjustedType, isIUO) = adjustTypeForConcreteImport(
+      *this, type, importResult.AbstractType, importKind, importResult.Hint,
+      allowNSUIntegerAsInt, bridging, optionality, resugarNSErrorPointer);
+
+  // We should never get an actual IUO type back.
+  assert(!adjustedType ||
+         !adjustedType->getImplicitlyUnwrappedOptionalObjectType());
+
+  // Make an IUO type based on isIUO. This will be removed when IUOs
+  // are removed from the type system.
+  adjustedType = adjustOptionality(adjustedType, isIUO);
+
+  return {adjustedType, isIUO};
+}
+
+Type ClangImporter::Implementation::importTypeIgnoreForceUnwrap(
```

Comment:
This should be "importTypeIgnoringIUO", right? "Force unwrap" is the operator `!`.

(I looked at your use sites and it seems like it's close to being `importTypeIgnoringNullability`, but I guess it's not that in practice. Too bad, because that's way easier to understand.)